content-services/content-sources-backend
# Fixes 1581: cancel in-progress snapshot on update

## Summary
This adds task cancellation to the tasking system and uses it to cancel in-progress snapshots when the URL is updated mid-snapshot.

I also added in a temporary task cancel API endpoint to use for testing while developing.

TODO
- [ ] potentially remaining TODO comments

## Testing steps
1. Create a repository with snapshots enabled. Use a larger one that takes some time to snapshot.
2. Immediately update that repository to use a different URL.
3. A new snapshot task will start for the different URL. The original snapshot will be canceled.
4. When the original snapshot task finishes, the status should be "canceled" and there should be no snapshot in the db or objects in pulp.
5. Query pulp with the hrefs on the original snapshot task payload to check that there are no objects in pulp. The tasks will exist, but they should have no created resources (repository version, publication, distribution).
6. The new snapshot task should be completed as normal.
